package/kernel/mac80211/patches/ath/407-regulatory-sanitize-extra-codes.patch

```diff
@@ -0,0 +1,18 @@
+Index: backports-4.19.7-1/drivers/net/wireless/ath/regd.c
+===================================================================
+--- backports-4.19.7-1.orig/drivers/net/wireless/ath/regd.c
++++ backports-4.19.7-1/drivers/net/wireless/ath/regd.c
+@@ -703,6 +703,12 @@ ath_regd_init_wiphy(struct ath_regulator
+  */
+ static void ath_regd_sanitize(struct ath_regulatory *reg)
+ {
++	/*
++	 * This will sanitize the following values:
++	 * - 0x834b -> in the Linksys EA6350v3
++	 * - 0x8000 -> some manufacturers use this invalid code
++	 */
+-	if (reg->current_rd != COUNTRY_ERD_FLAG)
++	if (reg->current_rd != COUNTRY_ERD_FLAG &&
++	    reg->current_rd != 0x834b)
```

Comment:
The regdomain consists of COUNTRY_ERD_FLAGs (WORLDWIDE_ROAMING_FLAG) flags and the country code.
In your case the country code  is 0x34b = 843 .

As luck would have it, I found this mail from @ecsv 
http://lists.infradead.org/pipermail/ath10k/2017-August/010014.html

That says maps 843 to CTRY_UNITED_STATES3. Could you please check if the following change instead of the ath_regd_sanitize patch helps?  
```
diff --git a/drivers/net/wireless/ath/regd.h b/drivers/net/wireless/ath/regd.h
index d73e45e26547..75ddaefdd049 100644
--- a/drivers/net/wireless/ath/regd.h
+++ b/drivers/net/wireless/ath/regd.h
@@ -185,7 +185,9 @@ enum CountryCode {
 	CTRY_UKRAINE = 804,
 	CTRY_UNITED_KINGDOM = 826,
 	CTRY_UNITED_STATES = 840,
+	CTRY_UNITED_STATES2 = 841,
 	CTRY_UNITED_STATES_FCC49 = 842,
+	CTRY_UNITED_STATES3 = 843,
 	CTRY_URUGUAY = 858,
 	CTRY_UZBEKISTAN = 860,
 	CTRY_VENEZUELA = 862,
--- drivers/net/wireless/ath/regd_common.h.orig	2019-01-04 15:55:01.733874390 +0100
+++ drivers/net/wireless/ath/regd_common.h	2019-01-04 15:58:00.533892825 +0100
@@ -486,6 +486,8 @@ static struct country_code_to_enum_rd al
 	{CTRY_UNITED_KINGDOM, ETSI1_WORLD, "GB"},
 	{CTRY_UNITED_STATES, FCC3_FCCA, "US"},
 	{CTRY_UNITED_STATES, FCC3_FCCA_2, "US"},
+	{CTRY_UNITED_STATES2, FCC3_FCCA, "US"},
+	{CTRY_UNITED_STATES3, FCC3_FCCA, "US"},
 	/* This "PS" is for US public safety actually... to support this we
 	 * would need to assign new special alpha2 to CRDA db as with the world
 	 * regdomain and use another alpha2 */
```

Reply:
Thanks for the review. I will abandon this PR and apply the changes. I see that I have to do some minor changes, mostly in format and to keep on line with the new changes I wasn't aware of.

But one thing that I don't understand is how to fix this. I mean, yes I know how to do it but the question is: shall I commit this change to the patch or it will be committed to master by you?

Because without it, I cannot test the code successfully.

Reply:
>Thanks for the review. I will abandon this PR and apply the changes. I see that I have to do some minor changes, mostly in format and to keep on line with the new changes I wasn't aware of.

But why abandon the PR? You can use the power of [git rebase](https://openwrt.org/submitting-patches#alternative_squashing_advice) to edit the patches and use git push force to update it.

> But one thing that I don't understand is how to fix this. I mean, yes I know how to do it but the question is: shall I commit this change to the patch or it will be committed to master by you?

Depends, have you tested it? Obviously, if it doesn't work then we stick to your solution.
If it does, you can just use the code for your commit.... And Ideally linux-wireless should be notified about the change... (However @ecsv already did back in 2017 and nobody cared :disappointed: ).

Reply:
Your patch, indeed, solved the problem. Thank you very much for providing a cleaner solution.
I'll add this code to my commit for mac80211. (shall I use "mac80211:" or "mac80211: ath:" or "ath"?)